src/rest/resource.cpp

```diff
@@ -800,6 +809,256 @@ void Resource::DatasetPending(const Request &aRequest, Response &aResponse) cons
+
+    addrPtr = &joiner.mSharedId.mEui64;
+    if (memcmp(&joiner.mSharedId.mEui64, emptyArray, OT_EXT_ADDRESS_SIZE) == 0)
+    {
+        addrPtr = nullptr;
+    }
+
+    if (joiner.mType == OT_JOINER_INFO_TYPE_DISCERNER)
+    {
+        errorOt = otCommissionerAddJoinerWithDiscerner(mInstance, &joiner.mSharedId.mDiscerner, joiner.mPskd.m8,
+                                                       joiner.mExpirationTime);
+    }
+    else
+    {
+        errorOt = otCommissionerAddJoiner(mInstance, addrPtr, joiner.mPskd.m8, joiner.mExpirationTime);
+    }
+    VerifyOrExit(errorOt == OT_ERROR_NONE, error = OTBR_ERROR_OPENTHREAD);
+
+exit:
+    if (error == OTBR_ERROR_NONE)
+    {
+        errorCode = GetHttpStatus(HttpStatusCode::kStatusOk);
+        aResponse.SetResponsCode(errorCode);
+    }
+    else if (error == OTBR_ERROR_INVALID_STATE)
+    {
+        ErrorHandler(aResponse, HttpStatusCode::kStatusConflict);
+    }
+    else if (error == OTBR_ERROR_INVALID_ARGS)
+    {
+        ErrorHandler(aResponse, HttpStatusCode::kStatusBadRequest);
+    }
+    else if (error == OTBR_ERROR_OPENTHREAD)
+    {
+        if (errorOt == OT_ERROR_INVALID_ARGS)
+        {
+            ErrorHandler(aResponse, HttpStatusCode::kStatusBadRequest);
+        }
+        else if (errorOt == OT_ERROR_NO_BUFS)
+        {
+            ErrorHandler(aResponse, HttpStatusCode::kStatusInsufficientStorage);
+        }
+        else
+        {
+            ErrorHandler(aResponse, HttpStatusCode::kStatusInternalServerError);
+        }
+    }
+    else
+    {
+        ErrorHandler(aResponse, HttpStatusCode::kStatusInternalServerError);
+    }
+}
+
+void Resource::RemoveJoiner(const Request &aRequest, Response &aResponse) const
+{
+    otbrError         error = OTBR_ERROR_NONE;
+    std::string       errorCode;
+    otExtAddress      eui64;
+    otExtAddress     *addrPtr   = nullptr;
+    otJoinerDiscerner discerner = {
+        .mValue  = 0,
+        .mLength = 0,
+    };
+    std::string body;
+
+    VerifyOrExit(otCommissionerGetState(mInstance) == OT_COMMISSIONER_STATE_ACTIVE, error = OTBR_ERROR_INVALID_STATE);
+
+    VerifyOrExit(Json::JsonString2String(aRequest.GetBody(), body), error = OTBR_ERROR_INVALID_ARGS);
+    if (body != "*")
+    {
+        error = Json::StringDiscerner2Discerner(const_cast<char *>(body.c_str()), discerner);
+        if (error == OTBR_ERROR_NOT_FOUND)
+        {
+            error = OTBR_ERROR_NONE;
+            VerifyOrExit(Json::Hex2BytesJsonString(body, eui64.m8, OT_EXT_ADDRESS_SIZE) == OT_EXT_ADDRESS_SIZE,
+                         error = OTBR_ERROR_INVALID_ARGS);
+            addrPtr = &eui64;
+        }
+        else if (error != OTBR_ERROR_NONE)
+        {
+            ExitNow(error = OTBR_ERROR_INVALID_ARGS);
+        }
+    }
+
+    if (discerner.mLength == 0)
+    {
+        otCommissionerRemoveJoiner(mInstance, addrPtr);
```

Comment:
The CI test is failing with this line, see https://github.com/openthread/ot-br-posix/actions/runs/11502608180/job/32019937960?pr=2515. This is because OpenThread API requires checking the returned `otError` value. If you are sure that this method can't fail, you can ignore it by adding the **(void)** prefix.

```suggestion
        (void) otCommissionerRemoveJoiner(mInstance, addrPtr);
```

There are more errors like this, you would want to fix all of them to pass the CI.